tests/profile/tracking/conftest.py

```diff
@@ -0,0 +1,108 @@
+#
+# conftest.py
+#
+# Copyright (C) 2017 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+#
+# Definition of fixtures for tracking tests
+#
+
+
+import os
+import json
+from uuid import uuid1, uuid5
+import pytest
+
+from kano.utils.file_operations import ensure_dir
+from kano_profile.tracker.tracking_session import TrackingSession
+from kano_profile.paths import tracker_dir, PAUSED_SESSIONS_FILE
+
+
+class TrackingSessionFixture(object):
+    @staticmethod
+    def format_session(name, started, pid, elapsed, finished):
+        return {
+            "name": name,
+            "started": started,
+            "pid": pid,
+            "elapsed": elapsed,
+            "finished": finished,
+            "app_session_id": str(uuid5(uuid1(), '{}{}'.format(name, pid)))
+        }
+
+    @classmethod
+    def setup_session_fixtures(cls, directory, sessions):
+        ensure_dir(directory)
+        for session_f in os.listdir(directory):
+            session_path = os.path.join(directory, session_f)
+            if os.path.isfile(session_path):
+                os.remove(session_path)
+
+        for session in sessions:
+            session_obj = TrackingSession(
+                name=session['name'],
+                pid=session['pid']
+            )
+            session_path = os.path.join(directory, session_obj.file)
+
+            with open(session_path, 'w') as session_f:
+                json.dump(session, session_f)
+
+
+    @classmethod
+    def setup_sessions(cls, sessions):
+        cls.setup_session_fixtures(tracker_dir, sessions)
+
+
+    @classmethod
+    def setup_paused_sessions(cls, sessions):
+        if sessions == None:
+            if os.path.exists(PAUSED_SESSIONS_FILE):
+                os.remove(PAUSED_SESSIONS_FILE)
+
+            return
+
+        with open(PAUSED_SESSIONS_FILE, 'w') as sessions_f:
+            for session in sessions:
+                sessions_f.write(
+                    '{}\n'.format(json.dumps({
+                        'name': session['name'],
+                        'pid': session['pid']
+                    }))
+                )
+
+
+@pytest.fixture(scope='function')
+def sample_tracking_sessions():
+    return [
```

Comment:
Rather than returning a list of objects here, you can parameterise the fixture with `@pytest.fixture(scope='function', params=(list_of_stuff))`. This triggers the test using the fixture to be re-run for each parameter of the fixture so you'll avoid doing multiple asserts for each list item.